### PR TITLE
Support windows generate `kyuubi-version-info.properties`

### PR DESCRIPTION
**Pull request by cxzl25**
### _Why are the changes needed?_
#3847

Windows users need to generate `kyuubi-version-info.properties` to debug.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: build/kyuubi-build-info.cmd [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+echo kyuubi_hadoop_version=%~7
+echo kyuubi_flink_version=%~8
+echo kyuubi_trino_version=%~9
+echo user=%username%
+
+FOR /F %%i IN ('git rev-parse HEAD') DO SET "revision=%%i"
+FOR /F %%i IN ('git rev-parse --abbrev-ref HEAD') DO SET "branch=%%i"
+FOR /F %%i IN ('git config --get remote.origin.url') DO SET "url=%%i"
+
+FOR /f %%i IN ('date /t') DO SET current_date=%%i
```
Comment (cxzl25) [inline build/kyuubi-build-info.cmd:42]: The use of `date /t` under cmd needs to consider the locale, so the return value of `date /t` is not parsed here

File: build/kyuubi-build-info.cmd [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+rem distributed under the License is distributed on an "AS IS" BASIS,
+rem WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+rem See the License for the specific language governing permissions and
+rem limitations under the License.
+rem
+
+SET EXTRA_RESOURCE_DIR=%~1
+IF NOT EXIST "%EXTRA_RESOURCE_DIR%" MKDIR "%EXTRA_RESOURCE_DIR%"
+SET BUILD_INFO="%EXTRA_RESOURCE_DIR%"\kyuubi-version-info.properties
+CALL :echo_build_properties %* > "%BUILD_INFO%"
```
Comment (cxzl25) [inline build/kyuubi-build-info.cmd:23]: windows

image

**Comment by wForget** [comment]
Great, run succeeded on windonws 10:
image

**Review by wForget — APPROVED** [review]

**Comment by pan3793** [comment]
Could you please test run test or debug in IDEA on Windows?

**Comment by wForget** [comment]
> Could you please test run test or debug in IDEA on Windows?

It can also run successfully.

image

**Comment by cxzl25** [comment]
> Could you please test run test or debug in IDEA on Windows?

Debug

image


Some UTs that do not require `HADOOP_HOME` can run directly in IDEA

image

**Review by pan3793 — APPROVED** [review]
Great! Thanks @cxzl25

**Pull request closed without merging**